### DISCUSSION TITLE
OOP2: why a closed source component ?

### DISCUSSION THREAD

**Discussion by marinthiercelin** [NightscoutFoundation/xDrip#3798] [Q&A]
Hello,

Is there any specific reason why the decoding of data from Freestyle Libre 2 sensors is handled by OOP2, an external process that is not open source ?

Is it an issue of licensing ? 
Would I get into legal trouble if I would reverse engineer the OOP2 service to make a similar service ?

Thanks for any advice

**Comment by PhilTU** [comment]
It allows rapid response to sensor changes without of xDrip code. It may contain elements of code that could be targetted by the OEM for IP violations. Two possible reasons for keeping it outside.
